source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -79,7 +80,11 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   Router::RouteConstSharedPtr route = callbacks_->route();
   skip_check_ = skipCheckForRoute(route);
 
-  if (!config_->filterEnabled() || skip_check_) {
+  bool disabled = !config_->filterEnabled(callbacks_->streamInfo().dynamicMetadata());
+  if (disabled) {
+    stats_.disabled_.inc();
+  }
+  if (disabled || skip_check_) {
     if (skip_check_) {
```

Comment:
I think the code would be easier to read if this condition was moved to the place of the newly introduced increment of `stats_.disabled_` (lines 84-86) and the increment was placed to the line 88:
```c++
if (skip_check_) {
  return Http::FilterHeadersStatus::Continue;
}

if (disabled) {
  stats_.disabled_.inc();
  if (config_->denyAtDisable()) {
    ...
  }
  ...
}
```

Reply:
+1

Reply:
thanks, updated.